pkg/analyze/kube_resource_test.go

```diff
@@ -0,0 +1,68 @@
+package analyzer
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	"github.com/pkg/errors"
+	troubleshootv1beta2 "github.com/replicatedhq/troubleshoot/pkg/apis/troubleshoot/v1beta2"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_clusterResource(t *testing.T) {
+	tests := []struct {
+		name     string
+		isError  bool
+		analyzer troubleshootv1beta2.ClusterResource
+	}{
+		{
+			name: "namespaced resource",
+			analyzer: troubleshootv1beta2.ClusterResource{
+				CollectorName: "Check namespaced resource",
+				Kind:          "Deployment",
+				Namespace:     "kube-system",
+				Name:          "coredns",
+			},
+		},
+		{
+			name: "cluster scoped resource",
+			analyzer: troubleshootv1beta2.ClusterResource{
+				CollectorName: "Check namespaced resource",
+				Kind:          "Node",
+				Name:          "repldev-marc",
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			req := require.New(t)
```

Comment:
Use `assert` package. `require` package [fails immediately an assertion is invalid](https://github.com/stretchr/testify#require-package). The package is also intended to check preconditions before starting a test e.g ensuring we have a DB connection etc.